src/indexable.cr

```diff
@@ -412,24 +412,80 @@ module Indexable(T)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block once for each index in `self`, passing `self`
+  # and `other` index element as parameter.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # b = ["a", "b", "c"]
+  #
+  # a.zip(b)  { |x,y| puts "[ #{x} -- #{y} ]" }
+  # ```
+  #
+  # produces:
+  #
+  # ```text
+  # [ 1 -- a ]
+  # [ 2 -- b ]
+  # [ 3 -- c ]
+  #```
   def zip(other : Indexable)
     each_with_index do |elem, i|
       yield elem, other[i]
     end
   end
 
+  # Returns an `Array` of Tuples populated with the *N* index
+  # element from `self` and *N* index element from `other`.
+  #
+  # Raises IndexError if `other` has no index corresponding to `self`.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # b = ["a", "b", "c"]
+  #
+  # a.zip(b) # => [{1, "a"}, {2, "b"}, {3, "c"}]
+  # ```
   def zip(other : Indexable(U)) forall U
     pairs = Array({T, U}).new(size)
     zip(other) { |x, y| pairs << {x, y} }
     pairs
   end
 
+  # Returns an `Array` of Tuples populated with the *N* element from `self`
+  # and *N* element from `other`.
+  # It adds *nil* when `other` returns outside the range for the given index.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # b = [4, 5]
+  #
+  # a.zip(b) # => [{1, 1}, {2, 2}, {3, nil}]
```

Comment:
This result is incorrect, with the given sample data it would be `[{1, 4}, {2, 5}, {3, nil}]`. But it would probably be best to adjust the values in `b`.